storage/pebble/registers.go

```diff
@@ -74,22 +89,55 @@ func (s *Registers) Store(
 		}
 	}
 
-	err := batch.Commit(pebble.Sync)
+	err = batch.Commit(pebble.Sync)
 	if err != nil {
 		return fmt.Errorf("failed to commit batch: %w", err)
 	}
 
 	return nil
 }
 
-// TODO: Finish implementation after deciding prefixes and/or badger use
-
 // LatestHeight Gets the latest height of complete registers available
 func (s *Registers) LatestHeight() (uint64, error) {
-	panic("not implemented")
+	return s.heightLookup(latestHeightKey())
 }
 
 // FirstHeight first indexed height found in the store, typically root block for the spork
 func (s *Registers) FirstHeight() (uint64, error) {
-	panic("not implemented")
+	return s.heightLookup(firstHeightKey())
+}
+
+func (s *Registers) heightLookup(key []byte) (uint64, error) {
+	res, closer, err := s.db.Get(key)
+	defer closer.Close()
+	if err != nil {
+		return 0, err
+	}
+	return binary.BigEndian.Uint64(res), nil
+}
+
+func (s *Registers) SetFirstHeight(height uint64) error {
+	return s.indexHeightWithKey(firstHeightKey(), height)
```

Comment:
First height should only be set once.

Reply:
correct, but we should still have this function for access in `access_node_builder`

Reply:
I mean we should not allow overwritten a value. We set first height during bootstrapping with the root height, and after that we won't set it, the caller (regardless access or execution) should be able to know whether pebble database has been bootstrapped with the first height, right? 

Importing the checkpoint is the bootstrapping phase, we will update the first height along with it.

Reply:
Could we encapsulate setting first height when we call `Store()` for first time?

Reply:
@sideninja @zhangchiqing the core reason I've implemented the above is for this case:
- The checkpoint is large, and will call `Store` concurrently over each sub-Trie: https://github.com/onflow/flow-archive/blob/master/service/mapper/transitions.go#L134-L151
- If `Store` updates `LatestHeight` or `FirstHeight` during the first batch and fails, and on re-starts reads the same pebble file, it will find a `LatestHeight` and `FirstHeight` already set. 

There's two other paths I've considered but don't want to go down yet:
1. Make a separate `FromCheckpoint` function in `Registers` that handles the atmoicity problem, and use the original implementation of `Store`
2. pass in `RootHeight` in `NewRegisters`  and set `LatestHeight` and `FirstHeight` and let `Store` update as normal, but now you'll have to re-design this if you want pruning etc.

I think what I want to do is 2. and then preserve the original behavior of `Store` if we want to remove the `Set` functions

Reply:
Can you use the Pebble store concurrently tho? cc @zhangchiqing

Reply:
I think we should create a `Import` method that takes a channel of registers, and store them concurrently, and after the channel is drained, set the first height.

This is where we concurrently set registers to pebble in archive node
https://github.com/onflow/flow-archive/blob/master/service/mapper/transitions.go#L183